core/imt/src/RTaskArena.cxx

```diff
@@ -140,10 +155,28 @@ std::shared_ptr<ROOT::Internal::RTaskArenaWrapper> GetGlobalTaskArena(unsigned m
       }
       return sp;
    }
-   std::shared_ptr<ROOT::Internal::RTaskArenaWrapper> sp(new ROOT::Internal::RTaskArenaWrapper(maxConcurrency));
+   std::shared_ptr<ROOT::Internal::RTaskArenaWrapper> sp;
+   if (semantic && config == ROOT::EIMTConfig::kExistingTBBArena) {
+      sp = std::make_shared<ROOT::Internal::RTaskArenaWrapper>(ROOT::Internal::RTaskArenaWrapper::Attach{});
+   } else {
+      if (semantic && config != ROOT::EIMTConfig::kWholeMachine) {
+         maxConcurrency = 0;
+      }
```

Comment:
I don't understand this branch, which other value for `config` is this testing for? And why is it setting `maxConcurrency = 0` (the whole machine) if `config != ROOT::EIMTConfig::kWholeMachine`? Did you mean
```suggestion
      if (semantic && config == ROOT::EIMTConfig::kWholeMachine) {
         maxConcurrency = 0;
      }
```